Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h

```diff
@@ -413,6 +418,58 @@ class CEXIETHERNET : public IEXIDevice
 #endif
   };
 
+  class BuiltInBBAInterface : public NetworkInterface
+  {
+  public:
+    BuiltInBBAInterface(CEXIETHERNET* eth_ref, std::string dns_ip)
+        : NetworkInterface(eth_ref), m_dns_ip(std::move(dns_ip))
+    {
+    }
+
+  public:
+    bool Activate() override;
+    void Deactivate() override;
+    bool IsActivated() override;
+    bool SendFrame(const u8* frame, u32 size) override;
+    bool RecvInit() override;
+    void RecvStart() override;
+    void RecvStop() override;
+
+  private:
+    std::string m_mac_id;
+    std::string m_dns_ip;
+    bool active = false;
+    u16 ip_frame_id = 0;
+    u8 queue_read = 0;
+    u8 queue_write = 0;
+    u16 queue_data_size[16]{};
+    std::unique_ptr<u8[]> queue_data[16];
+    std::mutex mtx;
+#if defined(WIN32) || defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) ||          \
+    defined(__OpenBSD__) || defined(__NetBSD__) || defined(__HAIKU__)
+    StackRef network_ref[10]{};  // max 10 at same time, i think most gc game had a limit of 8 in
+                                 // the gc framework
+    std::unique_ptr<u8[]> m_in_frame;
+    std::unique_ptr<u8[]> m_out_frame;
+    std::thread m_read_thread;
+    Common::Flag m_read_enabled;
+    Common::Flag m_read_thread_shutdown;
+    static void ReadThreadHandler(BuiltInBBAInterface* self);
+    Common::MACAddress fake_mac{};
+#endif
+    void WriteToQueue(u8* data, int length);
+    void HandleARP(Common::EthernetHeader* hwdata, Common::ARPHeader* arpdata);
+    void HandleDHCP(Common::EthernetHeader* hwdata, Common::UDPHeader* udpdata,
+                    Common::DHCPBody* request);
+    u8 GetAvaibleSlot(u16 port);
```

Comment:
Can't a pointer to a `StackRef` be returned (or `nullptr` if no results) instead? I don't think I've seen the index being used for anything meaningful beside accessing the `network_ref` array.